resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/rest/kubernetes/v2/ResourceStagingServerSuite.scala

```diff
@@ -39,30 +40,50 @@ import org.apache.spark.util.Utils
  * we've configured the Jetty server correctly and that the endpoints reached over HTTP can
  * receive streamed uploads and can stream downloads.
  */
-class ResourceStagingServerSuite extends SparkFunSuite with BeforeAndAfterAll {
+class ResourceStagingServerSuite extends SparkFunSuite with BeforeAndAfter {
+  private val OBJECT_MAPPER = new ObjectMapper().registerModule(new DefaultScalaModule)
 
   private val serverPort = new ServerSocket(0).getLocalPort
   private val serviceImpl = new ResourceStagingServiceImpl(Utils.createTempDir())
-  private val server = new ResourceStagingServer(serverPort, serviceImpl)
-  private val OBJECT_MAPPER = new ObjectMapper().registerModule(new DefaultScalaModule)
+  private val sslOptionsProvider = new SettableReferenceSslOptionsProvider()
+  private val server = new ResourceStagingServer(serverPort, serviceImpl, sslOptionsProvider)
 
-  override def beforeAll(): Unit = {
+  after {
+    server.stop()
+  }
+
+  test("Accept file and jar uploads and downloads") {
     server.start()
+    runUploadAndDownload(SSLOptions())
   }
 
-  override def afterAll(): Unit = {
-    server.stop()
+  test("Enable SSL on the server") {
+    val (keyStore, trustStore) = SSLUtils.generateKeyStoreTrustStorePair(
+      "127.0.0.1", "changeit", "changeit", "changeit")
```

Comment:
make these 3 values different so it's clear what's what and make sure the values a couple lines down are lined up